src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCompositeFormat.cs

```diff
@@ -0,0 +1,230 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.NetCore.Analyzers.Performance
+{
+    using static MicrosoftNetCoreAnalyzersResources;
+
+    /// <summary>
+    /// CA1862: <inheritdoc cref="UseCompositeFormatTitle"/>
+    /// </summary>
+    /// <remarks>
+    /// Roslyn already provides a refactoring for finding string.Format calls with literal string formats
+    /// and converting them to use string interpolation. This analyzer instead focuses on non-literal / const
+    /// arguments.
+    /// </remarks>
+    [DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+    public sealed class UseCompositeFormatAnalyzer : DiagnosticAnalyzer
+    {
+        internal static readonly DiagnosticDescriptor UseCompositeFormatRule = DiagnosticDescriptorHelper.Create("CA1862",
+            CreateLocalizableResourceString(nameof(UseCompositeFormatTitle)),
+            CreateLocalizableResourceString(nameof(UseCompositeFormatMessage)),
+            DiagnosticCategory.Performance,
+            RuleLevel.IdeSuggestion,
+            CreateLocalizableResourceString(nameof(UseCompositeFormatDescription)),
+            isPortedFxCopRule: false,
+            isDataflowRule: false);
+
+        internal const string StringIndexPropertyName = "StringIndex";
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(UseCompositeFormatRule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                INamedTypeSymbol stringType = compilationContext.Compilation.GetSpecialType(SpecialType.System_String);
+
+                // Get the types for CompositeFormat, IFormatProvider, and StringBuilder. If we can't, bail.
+                if (!compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextCompositeFormat, out INamedTypeSymbol? compositeFormatType) ||
+                    !compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemIFormatProvider, out INamedTypeSymbol? formatProviderType) ||
+                    !compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out INamedTypeSymbol? stringBuilderType))
+                {
+                    return;
+                }
+
+                // Process all calls to string.Format, assuming we can find all the members we'd use as replacements.
+                IMethodSymbol[] formatCompositeMethods = stringType.GetMembers("Format").OfType<IMethodSymbol>()
+                    .Where(m => m.IsStatic &&
+                                m.Parameters.Length >= 3 &&
```

Comment:
Why `>= 3`? The overload with 2 params like below could not covered?     
[`public static string Format([StringSyntax(StringSyntaxAttribute.CompositeFormat)] string format, object? arg0)`](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs#L422)

Reply:
Ah never mind, its for overloads having `CompositeFormat format` parameter